src/otx/algorithms/common/adapters/mmcv/semisl_mixin.py

```diff
@@ -0,0 +1,62 @@
+"""SemiSL configuration mixin."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import importlib
+
+from mmcv.utils import Config, ConfigDict
+
+from otx.algorithms.common.adapters.mmcv.utils import (
+    build_dataloader,
+    build_dataset,
+)
+
+
+class SemiSLConfigurerMixin:
+    """Patch config to support semi supervised learning."""
+
+    def configure_data(self, cfg, data_cfg):
+        """Patch cfg.data."""
+        super().configure_data(cfg, data_cfg)
+        # Set unlabeled data hook
+        if self.training:
+            if cfg.data.get("unlabeled", False) and cfg.data.unlabeled.get("otx_dataset", False):
+                if len(cfg.data.unlabeled.get("pipeline", [])) == 0:
```

Comment:
```suggestion
                if cfg.data.unlabeled.get("pipeline", []):
```
How about to align with PEP recommended way to check a empty list?

Reply:
I revert this change since it raises errors for segmentation semisl, and hangs for detection semisl.